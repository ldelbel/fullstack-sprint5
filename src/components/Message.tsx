import React, { ReactElement, useContext } from "react";
import MessageContext from "../contexts/MessageContext";
import { IMessageContext } from "../types/interfaces";
import "./Message.css";

function Message(): ReactElement<React.FC> {
  const { message, setMessage } = useContext<IMessageContext>(MessageContext);

  return message ? (
    <div className="alert">
      <span
        data-testid="close"
        className="closebtn"
        onClick={() => setMessage("")}
      >
        &times;
      </span>
      {message}
    </div>
  ) : (
    <div data-testid="empty"></div>
  );
}

export default Message;
