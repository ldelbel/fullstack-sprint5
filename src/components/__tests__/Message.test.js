import React from "react";
import { render } from "@testing-library/react";
import Message from "../Message";

jest.mock("react", () => {
  const react = jest.requireActual("react");
  return {
    ...react,
    useContext: jest.fn(),
  };
});

describe("Message component", () => {
  describe("when message state is not null", () => {
    it("should render component with messate state content", () => {
      React.useContext.mockImplementation(() => {
        return { message: "mensagem" };
      });
      const { getByText } = render(<Message />);
      expect(getByText("mensagem")).toBeInTheDocument();
    });

    it('should have a class of "alert"', () => {
      React.useContext.mockImplementation(() => {
        return { message: "mensagem" };
      });
      const { container } = render(<Message />);
      expect(container.firstChild).toHaveClass("alert");
    });

    it("should have a close button", () => {
      React.useContext.mockImplementation(() => {
        return { message: "mensagem" };
      });
      const { container } = render(<Message />);
      expect(container.firstChild.firstChild).toHaveClass("closebtn");
    });

    it("should match snapshot", () => {
      React.useContext.mockImplementation(() => {
        return { message: "mensagem" };
      });
      const message = render(<Message />);
      expect(message).toMatchSnapshot();
    });
  });

  describe("when message state is null", () => {
    it("should render an empty div", () => {
      React.useContext.mockImplementation(() => {
        return { message: null };
      });
      const { container } = render(<Message />);
      expect(container.firstChild).toBeEmptyDOMElement();
    });
  });
});
