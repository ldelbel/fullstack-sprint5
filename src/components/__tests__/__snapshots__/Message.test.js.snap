// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component when message state is not null should match snapshot 1`] = `
<div
  className="alert"
>
  <span
    className="closebtn"
    onClick={[Function]}
  >
    Ã—
  </span>
  mensagem
</div>
`;
